core/src/main/scala/cats/data/NonEmptyList.scala

```diff
@@ -352,7 +354,12 @@ private[data] sealed trait NonEmptyListInstances extends NonEmptyListInstances0
 
       def extract[A](fa: NonEmptyList[A]): A = fa.head
 
-      def traverse[G[_], A, B](fa: NonEmptyList[A])(f: A => G[B])(implicit G: Applicative[G]): G[NonEmptyList[B]] =
+      def traverse1[G[_] : Apply, A, B](as: NonEmptyList[A])(f: A => G[B]): G[NonEmptyList[B]] = {
+        as.map(a => Apply[G].map(f(a))(NonEmptyList.of(_)))
+          .reduceLeft((acc, a) => (acc |@| a).map( _ |+| _ ))
+      }
```

Comment:
This is not really the best implementation of `traverse1`.

A better implementation : 

```scala
def traverse1[G[_], A, B](nel: NonEmptyList[A])(f: A => G[B])(implicit G: Apply[G]): G[NonEmptyList[B]] =
  Foldable[List].reduceRightToOption[A, G[List[B]]](nel.tail)(a => G.map(f(a))(_ :: Nil)) { (a, lglb) =>
    G.map2Eval(f(a), lglb)(_ :: _)
  }.map { 
    case None => G.map(f(nel.head))(NonEmptyList(_, Nil))
    case Some(gtail) => G.map2(f(nel.head), gtail)(NonEmptyList(_, _))
  }.value
```

`Traverse[Vector].traverse` also uses prepending (`+:`), so I think this should work for `NonEmptyVector` as well. For `OneAnd` we need to come up with something else if we want to improve that one as well.

A [benchmark](https://gist.github.com/peterneyens/befa719c6b9c9ee92434106910734619) comparing your `traverse1` to mine above and a copy of yours using `NonEmptyList(_, Nil)` instead of `NonEmptyList.of`.

```
[info] Benchmark                          Mode  Cnt      Score     Error  Units
[info] NelTraverseBench._traverse1noOf   thrpt   10  10977.183 ± 240.703  ops/s
[info] NelTraverseBench._traverse1prime  thrpt   10  89787.940 ± 849.006  ops/s
[info] NelTraverseBench.traverse1        thrpt   10   3929.498 ±  72.846  ops/s
```
The difference of just replacing `NonEmptList.of` is already quite big, so I am plugging #1707 here again.